Added 'security requirements' corresponding to policies in place (#230)
bcgov/public-cloud-techdocs: docs/azure/azure-services/front-door.md

Azure Front Door is a global entry point that uses the Microsoft edge network to deliver fast, secure, and highly scalable web applications. It provides dynamic site acceleration (DSA), SSL offloading, global load balancing with instant failover, and application layer security.

## Security requirements

> The following security settings are **required** in Landing Zones.

Reviewer: ```suggestion
The following security settings are **required** in Landing Zones:
```

> - Azure Front Door profiles must use the **Premium tier** that supports managed WAF rules and private link

Reviewer: ```suggestion
- Azure Front Door profiles must use the **Premium tier** that supports managed Web Application Firewall (WAF) rules and private link
```

- Azure Front Door profiles must use a **minimum TLS version of 1.2**
- Web Application Firewall (WAF) on Azure Front Door must have **request body inspection** enabled
- Web Application Firewall (WAF) must be enabled for Azure Front Door **entry-points**
- **Bot Protection** must be enabled for Azure Front Door WAF
- Web Application Firewall (WAF) **rate limit rule** for Azure Front Door must be enabled

> - Secure **private connectivity** between Azure Front Door Premium and Azure Storage Blob, or Azure App Service must be configured

Reviewer: ```suggestion
- Secure **private connectivity** between Azure Front Door Premium and Azure Storage Blob or Azure App Service must be configured
```

> - Web Application Firewall (WAF) must use the specified **mode** (ie. **Prevention**) for Azure Front Door Service

Reviewer: ```suggestion
- WAF on Azure Front Door must have **request body inspection** enabled
- WAF must be enabled for Azure Front Door **entry-points**
- **Bot Protection** must be enabled for Azure Front Door WAF
- WAF **rate limit rule** for Azure Front Door must be enabled
- Secure **private connectivity** between Azure Front Door Premium and Azure Storage Blob or Azure App Service must be configured
- WAF must use the specified **mode** (i.e. **Prevention**) for Azure Front Door Service
```
